project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/BenchmarkAction.kt

```diff
@@ -0,0 +1,71 @@
+package org.plan.research.minimization.plugin
+
+import org.plan.research.minimization.plugin.benchmark.BenchmarkProject
+import org.plan.research.minimization.plugin.benchmark.BenchmarkResultAdapter
+import org.plan.research.minimization.plugin.errors.MinimizationError
+import org.plan.research.minimization.plugin.services.BenchmarkingService
+
+import com.intellij.notification.*
+import com.intellij.openapi.actionSystem.ActionUpdateThread
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.guessProjectDir
+
+class BenchmarkAction : AnAction() {
+    override fun getActionUpdateThread() = ActionUpdateThread.BGT
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project ?: return
+        if (!project.name.endsWith("dataset")) {
```

Comment:
Why?

Reply:
Basically was the second line I wrote, I don't think that's an important security barrier, 'll delete